vagrant/base.sh

```diff
@@ -20,38 +20,41 @@ set -ex
 # If you update this, also update tests/docker/Dockerfile
 export KIBOSH_VERSION=d85ac3ec44be0700efe605c16289fd901cfdaa13
 
-if [ -z `which javac` ]; then
-    apt-get -y update
-    apt-get install -y software-properties-common python-software-properties
-    add-apt-repository -y ppa:webupd8team/java
-    apt-get -y update
+path_to_jdk_cache() {
+  jdk_version=$1
+  echo "/var/cache/oracle-jdk8-installer/jdk-${jdk_version}-linux-64.tar.gz"
+}
 
-    # Try to share cache. See Vagrantfile for details
-    mkdir -p /var/cache/oracle-jdk8-installer
-    if [ -e "/tmp/oracle-jdk8-installer-cache/" ]; then
-        find /tmp/oracle-jdk8-installer-cache/ -not -empty -exec cp '{}' /var/cache/oracle-jdk8-installer/ \;
-    fi
-    if [ ! -e "/var/cache/oracle-jdk8-installer/jdk-8u171-linux-x64.tar.gz" ]; then
-      # Grab a copy of the JDK since it has moved and original downloader won't work
-      curl -s -L "https://s3-us-west-2.amazonaws.com/kafka-packages/jdk-8u171-linux-x64.tar.gz" -o /var/cache/oracle-jdk8-installer/jdk-8u171-linux-x64.tar.gz
-    fi
+fetch_jdk_tgz() {
+  jdk_version=$1
 
-    /bin/echo debconf shared/accepted-oracle-license-v1-1 select true | /usr/bin/debconf-set-selections
+  if [ ! -e $(path_to_jdk_cache $jdk_version) ]; then
+    mkdir -p $(dirname $(path_to_jdk_cache $jdk_version))
+    # Grab a copy of the JDK since it has moved and original downloader won't work
+    curl -s -L "https://s3-us-west-2.amazonaws.com/kafka-packages/jdk-${jdk_version}-linux-x64.tar.gz" -o $(path_to_jdk_cache $jdk_version)
+  fi
+}
 
-    # oracle-javaX-installer runs wget with a dot progress indicator which ends up
-    # as one line per dot in the build logs.
-    # To avoid this noise we redirect all output to a file that we only show if apt-get fails.
-    echo "Installing JDK..."
-    if ! apt-get -y install oracle-java8-installer oracle-java8-set-default >/tmp/jdk_install.log 2>&1 ; then
-        cat /tmp/jdk_install.log
-        echo "ERROR: JDK install failed"
-        exit 1
-    fi
+if [ -z `which javac` ]; then
+    apt-get -y update
+    apt-get install -y software-properties-common python-software-properties binutils gsfonts gsfonts-x11 java-common libfontenc1 libxfont1 x11-common xfonts-encodings xfonts-utils
```

Comment:
how did we come up with this list of packages? anything `x11` seems suspicious -- do we really need all of that since everything we run is headless? pulling in `x11` tends to result in bloated images.

Reply:
I had gotten the list from a job that was still installing via the webupd8 deb and copied the list of packages to be installed minus the installer packages themselves, not wanting to make too many changes at once (e.g. keep installing the same dependencies, just replace the installer package with our own install). I changed this to just java-common as I think that is probably something we need.